### PR TITLE
Add label creation and label association

### PR DESCRIPTION
**Pull request by bat3**

### CODE REVIEW

File: business/MetadataDatabase/Controllers/Labels/LabelController.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    public class LabelController : ControllerBase
+    {
+        private readonly ILabelService labelService;
+
+        public LabelController(ILabelService labelService)
+        {
+            this.labelService = labelService;
+        }
+
+        // GET: api/Label
```
Comment (cquennouelle) [inline business/MetadataDatabase/Controllers/Labels/LabelController.cs:22]: Label with a capital letter?

File: business/MetadataDatabase/Repository/Labels/LabelRepository.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    public class LabelRepository : MongoRepositoryBase<Label>, ILabelRepository
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LabelRepository"/> class.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        public LabelRepository(IOptions<MongoConfiguration> configuration) : base(configuration.Value)
+        {
+
```
Comment (cquennouelle) [inline business/MetadataDatabase/Repository/Labels/LabelRepository.cs]: Remove whiteline if not used

**Review by cquennouelle — COMMENTED** [review]
Another too technical review for me.

File: business/MetadataDatabase/Controllers/Labels/LabelController.cs (hunk 1 above)
Comment (bat3) [reply business/MetadataDatabase/Controllers/Labels/LabelController.cs:22]: Yes like Series and Users
image

File: business/MetadataDatabase/Repository/Labels/LabelRepository.cs (hunk 2 above)
Comment (bat3) [reply business/MetadataDatabase/Repository/Labels/LabelRepository.cs]: Done

File: business/MetadataDatabase/Controllers/Labels/LabelController.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        public ActionResult<IEnumerable<LabelDto>> GetAllLabels()
+        {
+            return this.Ok(this.labelService.GetAll());
+        }
+
+        // GET: api/Label/5
+        [HttpGet("{id}", Name = "GetLabel")]
+        public ActionResult<LabelDto> GetLabel(string id)
+        {
+            return this.labelService.Get(id);
```
Comment (ptitdam2001) [inline business/MetadataDatabase/Controllers/Labels/LabelController.cs]: you should test if label does not exist : 

`
var label = this.labelService.Get(id);
            if (label == null) {
                return NotFound();
            }
            return label;
`

File: business/MetadataDatabase/Services/Labels/LabelService.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            return this.labelRepository.GetAll().ToDto();
+        }
+
+        public void Update(string id, LabelDto objectToUpdate)
+        {
+            objectToUpdate.Id = id;
+            this.labelRepository.Update(objectToUpdate.ToModel());
+        }
+
+        public bool IsValid(LabelDto label)
```
Comment (ptitdam2001) [inline business/MetadataDatabase/Services/Labels/LabelService.cs]: what is its usage? Why don't we use for update or create methods?

File: client/web/src/components/labels/LabelCreation.tsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+  };
+
+  const onCancel = () => {
+    if(props.onCancel !== undefined) props.onCancel();
+  }
+
+  const onCreate = () => {
+    let newLabel: ILabel = {
+      'id': '',
+      'user': user.login,
```
Comment (ptitdam2001) [inline client/web/src/components/labels/LabelCreation.tsx]: why don't we use user.id ?

File: client/web/src/components/labels/LabelCreation.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+          key='Multi' 
+          value='Multi' 
+        >
+          Multi values
+        </MenuItem>
+        <MenuItem 
+          key='String' 
+          value='String' 
+        >
+          Free text
```
Comment (ptitdam2001) [inline client/web/src/components/labels/LabelCreation.tsx]: translation

File: client/web/src/features/labels/labelsService.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import { ISeries } from '../../models/series'
+
+import { ICrudService } from '../common/ICrudService'
+import { HttpRequest } from '../common/HttpRequest'
+import { ILabel } from '../../models/ILabel'
+
+export class LabelsService implements ICrudService<ILabel> {
+  private api: HttpRequest<ISeries>
```
Comment (ptitdam2001) [inline client/web/src/features/labels/labelsService.ts]: Why type ISeries?

File: client/web/src/store.ts [hunk 8]
```diff
@@ -5,16 +5,19 @@ import createSagaMiddleware from 'redux-saga'
 import { UserReducer, userSaga } from './features/user'
 import { seriesSaga } from './features/series'
 import { mirrorPacsSaga } from './features/mirrorPacs'
+import { labelsSaga } from './features/labels'
 import serieReducer from './features/series/seriesReducer'
 import { authSaga, AuthReducer } from './features/auth'
 import { VersionReducer, versionSaga } from './features/version'
+import labelsReducer from './features/labels/labelsReducer'
```
Comment (ptitdam2001) [inline client/web/src/store.ts]: you can move it to `import { labelsSaga } from './features/labels'`

**Review by ptitdam2001 — CHANGES_REQUESTED** [review]

File: business/MetadataDatabase/Controllers/Labels/LabelController.cs (hunk 3 above)
Comment (bat3) [reply business/MetadataDatabase/Controllers/Labels/LabelController.cs]: Yes done

File: business/MetadataDatabase/Services/Labels/LabelService.cs (hunk 4 above)
Comment (bat3) [reply business/MetadataDatabase/Services/Labels/LabelService.cs]: Yes, I will use it in create and update

File: client/web/src/components/labels/LabelCreation.tsx (hunk 5 above)
Comment (bat3) [reply client/web/src/components/labels/LabelCreation.tsx]: I was more readable duing my tests ^^
But id can be undefined...

File: client/web/src/features/labels/labelsService.ts (hunk 7 above)
Comment (bat3) [reply client/web/src/features/labels/labelsService.ts]: Bad copy past

File: client/web/src/store.ts (hunk 8 above)
Comment (bat3) [reply client/web/src/store.ts]: Done

File: client/web/src/components/labels/LabelCreation.tsx (hunk 5 above)
Comment (ptitdam2001) [reply client/web/src/components/labels/LabelCreation.tsx]: when user id can be undefined at creation?

**Review by ptitdam2001 — DISMISSED** [review]

File: client/web/src/components/labels/LabelCreation.tsx (hunk 5 above)
Comment (bat3) [reply client/web/src/components/labels/LabelCreation.tsx]: In the interface, i can change it 
`export interface IUser {  
  id?: string`

Comment (ptitdam2001) [reply client/web/src/components/labels/LabelCreation.tsx]: ah ok yes you can, but we have to verify registration is ok

Comment (bat3) [reply client/web/src/components/labels/LabelCreation.tsx]: Done

**Review by ptitdam2001 — APPROVED** [review]

**Review by ptitdam2001 — APPROVED** [review]

**Pull request merged**